File: Echo.Desktop/ContentLoader.fs
namespace MGNamespace

open Microsoft.Xna.Framework.Content
open System.Collections.Generic
open Microsoft.Xna.Framework.Graphics
open System.IO

type ContetLoader () =
    interface IContentLoader with
        member this.LoadTextures(contentManager: ContentManager): Dictionary<string,Texture2D> = 
            let result = Dictionary<string,Texture2D>()
            let files = Directory.GetFiles("Content/Textures")
            
            files
            |> Seq.iter(fun f-> 
                    let key = Path.GetFileNameWithoutExtension f
                    let path = sprintf "Textures/%s" key
                    result.Add(key, contentManager.Load<Texture2D>(path))
                )
            |> ignore

            result

File: Echo.Android/ContentLoader.cs
using System.Collections.Generic;
using System.IO;
using Android.Content.Res;
using Echo.Common;
using Microsoft.Xna.Framework.Content;
using Microsoft.Xna.Framework.Graphics;

namespace Echo.Android
{
    class ContentLoader : IContentLoader
    {
        private readonly AssetManager AssetManager;

        public ContentLoader(AssetManager AssetManager)
        {
            this.AssetManager = AssetManager;
        }

        public Dictionary<string, Texture2D> LoadTextures(ContentManager value)
        {
            var result = new Dictionary<string, Texture2D>();

            var textures = AssetManager.List("Content/Textures");

            foreach (var texture in textures)
            {
                var key = Path.GetFileNameWithoutExtension(texture);
                result.Add(key, value.Load<Texture2D>($"Textures/{key}"));
            }            

            return result;
        }
    }
}